corehq/apps/app_manager/fields.py

```diff
@@ -0,0 +1,93 @@
+import collections
+from django import forms
+from django.utils.translation import ugettext as _
+from corehq.apps.app_manager.models import get_apps_in_domain
+
+
+ApplicationDataSource = collections.namedtuple('ApplicationDataSource', ['application', 'source_type', 'source'])
+
+
+class ApplicationDataSourceWidget(forms.MultiWidget):
```

Comment:
bit of django doc digging and debugging to get something workable in these two classes